clearwater-cassandra/usr/share/clearwater/clearwater-cluster-manager/plugins/cassandra_failed_plugin.py

```diff
@@ -0,0 +1,60 @@
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2015 Metaswitch Networks Ltd
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+from metaswitch.clearwater.cluster_manager.plugin_base import SynchroniserPluginBase
+from metaswitch.clearwater.etcd_shared.plugin_utils import run_command
+
+class CassandraFailedPlugin(SynchroniserPluginBase):
+    def __init__(self, key, ip):
+        self._key = key
+        self._ip = ip
+
+    def key(self):
+        return self._key
+
+    def on_cluster_changing(self, cluster_view):
+        pass
+
+    def on_joining_cluster(self, cluster_view):
+        pass
+
+    def on_new_cluster_config_ready(self, cluster_view):
+        pass
+
+    def on_stable_cluster(self, cluster_view):
+        pass
+
+    def on_leaving_cluster(self, cluster_view):
+        # We must remove the node from the cassandra cluster. Get the node's ID
+        # from nodetool status, then remove it with nodetool remove
+        command = "nodetool removenode $(nodetool status | grep " + self._ip + " | sed -e's/ \+/ /g' | cut -f 7 -d ' ')"
```

Comment:
Given that this is a python script I'd probably have just run `nodetool status` then parsed the output in python. This would allow you parse the output more reliably, for example picking out the part of the line that looks like a UUID, rather than relying on the spacing in `nodetool status`. It would also have allowed you to log what was going on - at present we don't get to to see the result of the inner command. 

Having said that if changing this code is going to cause a costly retest its not worth doing. 

Either way, I would like there to be a comment showing the expected output of `nodetool status`. This makes it easier for someone reading the code to figure out what the shell command does.